ts/features/bonus/cgn/hooks/useCgnDiscountDetailBottomSheet.tsx

```diff
@@ -10,13 +11,29 @@ import {
   CgnDiscountDetailHeader
 } from "../components/merchants/CgnDiscountDetail";
 
+const calculateBottomSheetHeight = (
+  discount: Discount,
+  merchantType?: DiscountCodeType
+): number =>
+  310 +
+  (discount.description === undefined
+    ? 0
+    : discount.description.length * 0.75) +
+  (discount.condition === undefined ? 0 : discount.condition.length * 0.75) +
+  (merchantType === undefined ? -50 : 0);
```

Comment:
```suggestion
const screenHeight = Dimensions.get("screen").height;
const calculateBottomSheetHeight = (
  discount: Discount,
  merchantType?: DiscountCodeType
): number =>
  Math.min(
    310 +
      ((discount.description?.length ?? 0) +
        (discount.condition?.length ?? 0)) *
        0.75 +
      (merchantType ? 0 : -50),
    screenHeight
  );
```
I'm proposing a refactoring + limit the bottomsheet height to the device height
Could it be an improvement?

(bottomsheet overflow - I force the height to 1600)

https://user-images.githubusercontent.com/822471/152202642-14c54cc5-bb30-4cd3-aaf2-a2b17131a24e.mov

Reply:
Absolutely! thanks for the suggestion! c3b0b91 to solve!